high_card.rb

```diff
@@ -1,5 +1,95 @@
 # Use these two arrays to generate a deck of cards.
-ranks = ["A", 2, 3, 4, 5, 6, 7, 8, 9, 10, "J", "Q", "K" ]
-suits = [ "hearts", "spades", "clubs", "diamonds" ]
+ranks = ['A', 2, 3, 4, 5, 6, 7, 8, 9, 10, 'J', 'Q', 'K']
+suits = %w(Hearts Spades Clubs Diamonds)
+replay_hand = true  # To allow for replaying upon a tie
+
+deck = []
+
+ranks.each do |rank|
+  suits.each do |suit|
+    deck.push([rank, suit])
+  end
+end
+
+deck.shuffle!
 
 players = []
+user_input = ''
+
+while user_input != 'play'
+  print "#{players.length} players so far. Enter a player name or enter 'play': "
+  user_input = gets.chomp
+  if user_input.length > 0 && user_input != 'play'
+    players.push({name: user_input, rank: 0})
+  end
+end
+
+# players.push({name: "Dick", rank: 0})
+# players.push({name: "Jane", rank: 0})
+# players.push({name: "Buck", rank: 0})
+
+winner = {
+  name: '',
+  rank: 0
+}
+
+def get_rank(card_value)
+  rank = nil
+  case card_value
+  when (2..10)
+    rank = card_value - 1
+  when 'J'
+    rank = 10
+  when 'Q'
+    rank = 11
+  when 'K'
+    rank = 12
+  when 'A'
+    rank = 13
+  end
+  return rank
+end
+
+#
+# Not a terribly elegant solution for Bonus 1 and 2
+#
+
+while replay_hand do  # Loop in case of tie
```

Comment:
Great use of while loop and setting the conditional in the right spot. How could you import game config settings into your high card program using a config module?